enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_arr = []
+    my_each { |i| my_arr.push(i) if yield(i) }
+    my_arr
+  end
+
+  def my_all?(*args)
+    condition = true
+    if !args[0].nil?
+      my_each { |_i| condition = false }
+    elsif !block_given?
+      my_each { |_i| condition = false }
+    else
+      my_each { |i| condition = false unless yield(i) }
+    end
+    condition
+  end
+
+  def my_any?(*args)
+    state = false
+    if !args[0].nil?
+      my_each { |i| state = true if args[0] === i }
+    elsif !block_given?
+      my_each { |i| state = true if i }
+    else
+      my_each { |i| state = true if yield(i) }
+    end
+    state
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    state = true
+    my_each { |i| state = false if yield(i) }
+    state
+  end
+
+  def my_count
+    return to_enum(:my_count) unless block_given?
+
+    total = 0
+    my_each do |i|
+      total += 1 if block_given? && yield(i)
+      total = size unless block_given?
+    end
+    total
+  end
```

Comment:
Consider the possibility of not giving a block or argument, and just count the elements inside the array. Right now you are returning an enumerator not a number because you have unless block_given?.

Consider the possibility of receiving a number as an argument and count the equal elements in your enumerator.